draft-ietf-spice-sd-cwt.md

```diff
@@ -114,78 +114,161 @@ However the following guidance is generally recommended, regardless of protocol
+replaced with a dict containing only the special key "...".
 
 ~~~ cddl
-redacted_element = { "...": bstr }
+redacted_element = { "...": any }
 ~~~
 
-A Holder key binding CWT (#kbt) MUST be present in a `sd_kbt` claim in the unprotected header when presenting an SD-CWT to a Verifier.
-The `sd_kbt` claim can only be absent when the Issuer is providing the
-SD-CWT to the Holder.
+Blinded claims can be nested. For example, both individual keys in the
+`address` claim, and the entire `address` element can be separately blinded.
+An example nested claim is shown in **TODO iref**.
 
-An SD-CWT is a CWT containing zero or more Digested Salted Disclosed Claim, and zero or more Salted Disclosed Claims.
-The salt acts as a blinding factor, preventing a Verifier of an SD-CWT from learning claims that were not intentionally disclosed by a Holder.
+Finally, an issuer may create "decoy digests" which look like a blinded
+claim hash but have only a salt. Decoy digests are discussed in **TODO iref**.
 
-The following informative CDDL is provided to explain the syntax for an
-SD-CWT presentation. A complete CDDL schema is in (#cddl).
 
-Please note this example contains claims for demonstration of the disclosure syntax, such as `swversion`, `address`, and ``.
+# SD-CWT Issuance
+
+How the Holder communicates to the Issuer to request a CWT or an SD-CWT is
+out-of-scope of this specification. Likewise, how the Holder determines
+which claims to blind or to always disclose is a policy matter which is
+not discussed in this specification. This specification defines the format
+of an SD-CWT communicated between an Issuer and a Holder in this section, and describes the format of an SD-CWT communicated between a Holder and a
+Verifier in the next section.
+
+The unprotected header MUST contain an `sd_claims` section with a Salted
+Disclosed Claim for *every* blinded claim hash present anywhere in the
+payload. The payload MUST also include a key confirmation element (`cnf`)
+for the Holder's public key, and an `sd_alg` claim identifying the
+algorithm used to hash the Salted Disclosed Claims.
+
+## Issuer generation
+
+The Issuer follows all the requirements of generating a valid CWT.
+The Issuer MUST implement COSESign1 using an appropriate asymmetric
+signature algorithm / curve combination (for example ES256/P-256 or
+EdDSA/Ed25519)
+
+The Issuer MUST generate a unique cryptographically random salt of at least
+128-bits of entropy for each Salted Disclosed Claim. If the client
+communicates a client-generated nonce (`cnonce`) when requesting the SD-CWT,
+the Issuer SHOULD include it in the payload.
+
+## Holder validation
+
+Upon receiving an SD-CWT from the Issuer with the Holder as the subject, the
+Holder verifies the following:
+
+- the issuer (`iss`) and subject (`sub`) are correct;
+- if an audience (`aud`) is present, it is acceptable;
+- the CWT is valid according to the `nbf` and `exp` claims;
```

Comment:
```suggestion
- the CWT is valid according to the `iat` and `exp` claims;
```

Reply:
I meant `nbf` and `exp`. The current time is greater or equal to `nbf` and not more than `exp`. What comparison would you make with `iat`?

Reply:
Marking as resolved, to be addressed in Issue #14